Xamarin.Forms.Xaml/XamlLoader.cs

```diff
@@ -127,10 +131,14 @@ static void Visit (RootNode rootnode, HydratationContext visitorContext)
 
 		static string GetXamlForType(Type type)
 		{
-			string xaml = null;
-
 			//the Previewer might want to provide it's own xaml for this... let them do that
-			if (Xamarin.Forms.Xaml.Internals.XamlLoader.XamlFileProvider != null && (xaml = Xamarin.Forms.Xaml.Internals.XamlLoader.XamlFileProvider(type)) != null)
+			//the check at the end is preferred (using ResourceLoader). keep this until all the previewers are updated
+
+#pragma warning disable 0618
+			var xaml = Internals.XamlLoader.XamlFileProvider?.Invoke(type);
+#pragma warning restore 0618
+
+			if (xaml != null && ResourceLoader.ResourceProvider != null)
 				return xaml;
 
```

Comment:
Why are you checking that `ResourceProvider != null` here? I presume the idea is that we will *either* be using `XamlFileProvider` *or* we'll be using `ResourceLoader.ResourceProvider` which means if xaml is non-null then ResourceProvider will always be null.